code/__DEFINES/cooldowns.dm

```diff
@@ -72,3 +72,23 @@
 #define COOLDOWN_RESET(cd_source, cd_index) cd_source.cd_index = 0
 
 #define COOLDOWN_TIMELEFT(cd_source, cd_index) (max(0, cd_source.cd_index - world.time))
+
+
+
+// when a timer should track targets individually
+// this is useful when an item should individually trigger cooldown time per mob
+#define COOLDOWN_LIST_DECLARE(cd_index) var/list/##cd_index = list()
+
+#define COOLDOWN_STATIC_LIST_DECLARE(cd_index) var/static/list/##cd_index = list()
+
+/// IMPORTANT: "cd_target" should be individual because it's assoc key
+#define COOLDOWN_LIST_START(cd_target, cd_source, cd_index, cd_time) (cd_source.cd_index[cd_target] = world.time + (cd_time))
+
+#define COOLDOWN_LIST_FINISHED(cd_target, cd_source, cd_index) (cd_source.cd_index[cd_target] < world.time)
```

Comment:
These macro param names are confusing considering the index is actually the name of the cooldown and the target is the index.
The order of arguments should be:
object, cooldown name, index (cd_target), value (time/length)

Reply:
I changed macro param names. would that work?